src/recon_buildblock/PLSPrior.cxx

```diff
@@ -62,8 +62,16 @@ PLSPrior<elemT>::initialise_keymap()
 
 template <typename elemT>
 Succeeded
-PLSPrior<elemT>::set_up ()
+PLSPrior<elemT>::set_up (shared_ptr<DiscretisedDensity<3,elemT> > const& target_sptr)
 {
+    base_type::set_up(target_sptr);
+
+    if (kappa_filename.size() != 0)
```

Comment:
I have thought about this more, and I don't think reading the data here is a good idea. It means that calling `set_up` twice will read the data twice, and that errors when reading will only be flagged when doing the set_up. Furthermore, it introduces surprising behaviour (in my opinion) related to what to do with the filename vs the member.

So, please move the reading to the `set_kappa_filename` function, and call it from `post_processing`